DataManagementSystem/Agent/RequestOperations/MoveReplica.py

```diff
@@ -7,201 +7,324 @@
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RemoveReplicaOK", "Successful replica removals",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
-    gMonitor.registerActivity( "RemoveReplicaFail", "Failed replica removals",
-                               "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM )
+    super(MoveReplica, self).__init__(operation, csPath)
 
     # Init ConsistencyInspector: used to check replicas
     self.ci = ConsistencyInspector()
 
-  def __call__( self ):
+  def __call__(self):
     """ call me maybe """
+
+    # Check whether the ES flag is enabled so we can send the data accordingly.
+    if self.rmsMonitoring:
+      self.rmsMonitoringReporter = MonitoringReporter(monitoringType="RMSMonitoring")
+    else:
+      # # own gMonitor stuff for files
+      gMonitor.registerActivity("ReplicateAndRegisterAtt", "Replicate and register attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateOK", "Replications successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("ReplicateFail", "Replications failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterOK", "Registrations successful",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RegisterFail", "Registrations failed",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaAtt", "Replica removals attempted",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaOK", "Successful replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+      gMonitor.registerActivity("RemoveReplicaFail", "Failed replica removals",
+                                "RequestExecutingAgent", "Files/min", gMonitor.OP_SUM)
+
     # # check replicas first
     res = self.__checkReplicas()
     if not res["OK"]:
-      self.log.error( 'Failed to check replicas', res["Message"] )
+      self.log.error('Failed to check replicas', res["Message"])
 
     sourceSE = self.operation.SourceSE if self.operation.SourceSE else None
     if sourceSE:
       # # check source se for read
-      bannedSource = self.checkSEsRSS( sourceSE, 'ReadAccess' )
+      bannedSource = self.checkSEsRSS(sourceSE, 'ReadAccess')
       if not bannedSource["OK"]:
-        gMonitor.addMark( "ReplicateAndRegisterAtt", len( self.operation ) )
-        gMonitor.addMark( "ReplicateFail", len( self.operation ) )
+        if self.rmsMonitoring:
+          for opFile in self.operation:
```

Comment:
I do not like these status. The status is not `FileFailed` or `FileAttempted`. The status is `Attempted` or `Failed`. You have the `objectType` to specify `File`

Reply:
Okay, noted. I will work on these changes. And will change at every similar occurrence.